processor/tailsamplingprocessor/processor.go

```diff
@@ -59,6 +60,9 @@ type tailSamplingSpanProcessor struct {
 	nonSampledIDCache cache.Cache[bool]
 	deleteChan        chan pcommon.TraceID
 	numTracesOnMap    *atomic.Uint64
+
+	setPolicyMux  sync.RWMutex
```

Comment:
I didn't notice a read lock anywhere. Can this just be a `sync.Mutex`?
```suggestion
	setPolicyMux  sync.Mutex
```

Reply:
Yes, ended up w/ a single writer and reader 👍